Codata constants. (#800)
fortran-lang/stdlib: doc/specs/stdlib_constants.md

---
title: constants
---

[TOC]

## Introduction

The [[stdlib_constants]] module provides mathematical constants and the most common physical constants.

> The [[stdlib_codata]] module defines all codata constants as derived type [[codata_constant]].

Reviewer: DT have usually the suffix `_type` in `stdlib`:
```suggestion
The [[stdlib_codata]] module defines all codata constants as derived type [[codata_constant_type]].
```

> The lastest codata constants were released in 2018 by the [NIST](http://physics.nist.gov/constants.)

Reviewer: ```suggestion
The latest codata constants were released in 2018 by the [NIST](http://physics.nist.gov/constants.)
```

The names are quite long and can be aliased with shorter names.

> 

Reviewer: The DT should be described in details here. There should be also a note that all provided values are `dp`.

---
title: constants
---

[TOC]

## Introduction

The [[stdlib_constants]] module provides mathematical constants and the most common physical constants.

## Codata

The [[stdlib_codata(module)]] module defines all codata (physical) constants as derived 
type. The module is automatically generated with a simple 
[parser written in Python](https://github.com/MilanSkocic/codata/)
The latest codata constants were released in 2022 by the [NIST](http://physics.nist.gov/constants)
All values for the codata constants are provided as double precision reals. 
The names are quite long and can be aliased with shorter names.

The derived type [[stdlib_codata_type(module):codata_constant_type(type)]] defines:

* 4 members:

    * `name` (string)
    * `value` (double precision real)
    * `uncertainty` (double precision real)
    * `unit` (string)

* 2 type-bound procedures:

>     * `print` for print the values of the constant members.

Reviewer: ```suggestion
    * `print`: to print the values of the constant members;
```

>     * `to_real` for getting the value or the uncertainty to the desired precision.

Reviewer: ```suggestion
    * `to_real`: to get the value or the uncertainty to the desired precision.
```

A module level interface [[stdlib_codata_type(module):to_real(interface)]] is 
available for getting the constant value or uncertainty of a constant. 

> ## `to_real` - Get the constant value or uncertainty.

Reviewer: ```suggestion
## `to_real` - Get the constant value or its uncertainty.
```

### Status

Experimental

### Description

> Convert a [[stdlib_codata_type(module):codata_constant_type(type)]] to a real. 

Reviewer: ```suggestion
Convert a [[stdlib_codata_type(module):codata_constant_type(type)]] to a `real` (at least `sp`, or `dp`) scalar. 
```

> **Warning**: Some constants cannot be converted to simple precision reals due to the values of the exponents.

Reviewer: ```suggestion
**Warning**: Some constants cannot be converted to single precision `sp` reals due to the value of the exponents.
```
